src/utilities/controllers/turbine_controller.cpp

```diff
@@ -8,24 +8,25 @@
 namespace openturbine::util {
 
 TurbineController::TurbineController(
-    const std::string& shared_lib_path, const std::string& controller_function_name,
-    const std::string& input_file_path, const std::string& output_file_path
+    std::string shared_lib_path, std::string controller_function_name, std::string input_file_path,
+    std::string output_file_path
 )
-    : input_file_path_(input_file_path),
-      output_file_path_(output_file_path),
-      shared_lib_path_(shared_lib_path),
-      controller_function_name_(controller_function_name),
-      lib_(shared_lib_path, util::dylib::no_filename_decorations) {
+    : io{},
+      input_file_path_(std::move(input_file_path)),
```

Comment:
So is is the idea with `std::move` that `input_file_path` is already a copy of the original argument which will be moved into `input_file_path_`? Is the compiler not smart enough to figure this out itself?

Reply:
Yes, this is the idea.  And no, in most cases it is not.  In general, there's no promise that copy and move constructors don't have side effects, so the compiler is going to be limited in its ability to just pick the right one.  "Copy-and-move" has become pretty idiomatic modern C++ initialization, hence why Clang-Tidy checks for it.